aws-ts-eks-migrate-nodegroups/index.ts

```diff
@@ -0,0 +1,83 @@
+import * as awsx from "@pulumi/awsx"; import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import * as pulumi from "@pulumi/pulumi";
+import * as echoserver from "./echoserver";
+import * as iam from "./iam";
+import * as nginx from "./nginx";
+import * as utils from "./utils";
+
+const projectName = pulumi.getProject();
+
+// Allocate a new VPC with custom settings, and a public & private subnet per AZ.
+const vpc = new awsx.ec2.Vpc(`${projectName}`, {
+    cidrBlock: "172.16.0.0/16",
+    subnets: [{ type: "public" }, { type: "private" }],
+});
+
+// Export VPC ID and Subnets.
+export const vpcId = vpc.id;
+export const allVpcSubnets = vpc.privateSubnetIds.concat(vpc.publicSubnetIds);
+
+// Create 3 IAM Roles and matching InstanceProfiles to use with the nodegroups.
+const roles = iam.createRoles(projectName, 3);
+const instanceProfiles = iam.createInstanceProfiles(projectName, roles);
+
+// Create an EKS cluster.
+const myCluster = new eks.Cluster(`${projectName}`, {
+    version: "1.13",
+    vpcId: vpcId,
+    subnetIds: allVpcSubnets,
+    nodeAssociatePublicIpAddress: false,
+    skipDefaultNodeGroup: true,
+    deployDashboard: false,
+    instanceRoles: roles,
+    enabledClusterLogTypes: ["api", "audit", "authenticator",
+        "controllerManager", "scheduler"],
+});
+export const kubeconfig = myCluster.kubeconfig;
+export const clusterName = myCluster.core.cluster.name;
+
+// Create a Standard node group of t2.medium workers.
+const ngStandard = utils.createNodeGroup(`${projectName}-ng-standard`, {
+    ami: "ami-03a55127c613349a7", // k8s v1.13.7 in us-west-2
```

Comment:
we should add a call to `aws.getAmi` for this - this will mean we are able to be able to run our tests elsewhere easily - locally I use eu-west-1 as it's better latency for me

Reply:
There currently is no good, programmatic way of retrieving these AMIs from AWS. https://github.com/aws/containers-roadmap/issues/231 has served as a catch all to capture, and track this need from AWS.